Library/ViewModels/ChangeEmailViewModelTests.swift

```diff
@@ -0,0 +1,174 @@
+  private let dismissKeyboard = TestObserver<(), NoError>()
+  private let emailText = TestObserver<String, NoError>()
+  private let onePasswordButtonHidden = TestObserver<Bool, NoError>()
+  private let onePasswordFindLoginForURLString = TestObserver<String, NoError>()
+  private let passwordText = TestObserver<String, NoError>()
+  private let resendVerificationStackViewIsHidden = TestObserver<Bool, NoError>()
+  private let saveButtonIsEnabled = TestObserver<Bool, NoError>()
+
+  override func setUp() {
+    super.setUp()
+
+    self.vm.outputs.didChangeEmail.observe(self.didChangeEmail.observer)
+    self.vm.outputs.didFailToChangeEmail.observe(self.didFailToChangeEmail.observer)
+    self.vm.outputs.emailText.observe(self.emailText.observer)
+
+    self.vm.outputs.dismissKeyboard.observe(self.dismissKeyboard.observer)
+    self.vm.outputs.onePasswordButtonIsHidden.observe(self.onePasswordButtonHidden.observer)
+    self.vm.outputs.onePasswordFindLoginForURLString.observe(self.onePasswordFindLoginForURLString.observer)
+    self.vm.outputs.passwordText.observe(self.passwordText.observer)
+    self.vm.outputs.resendVerificationStackViewIsHidden.observe(
+      self.resendVerificationStackViewIsHidden.observer
+    )
+    self.vm.outputs.saveButtonIsEnabled.observe(self.saveButtonIsEnabled.observer)
+  }
+
+  func testDidChangeEmailEmits_OnSuccess() {
+
+    self.vm.inputs.passwordFieldDidTapGo(newEmail: "ksr@kickstarter.com", password: "123456")
+    self.scheduler.advance()
+
+    self.didChangeEmail.assertDidEmitValue()
+  }
+
+  func testDidFailToChangeEmailEmits_OnFailure() {
+
+    let error = GraphError.emptyResponse(nil)
+
+    withEnvironment(apiService: MockService(changeEmailError: error)) {
+
+      self.vm.inputs.passwordFieldDidTapGo(newEmail: "ksr@kickstarter.com", password: "123456")
+      self.scheduler.advance()
+
+      self.didFailToChangeEmail.assertDidEmitValue()
+    }
+  }
+
+  func testOnePasswordButtonHidesIfNotAvailable() {
+    self.vm.inputs.viewDidLoad()
+    self.vm.inputs.onePassword(isAvailable: false)
+
+    self.onePasswordButtonHidden.assertValues([true])
+  }
+
+  func testOnePasswordButtonVisibleIfAvailable() {
+
+    self.vm.inputs.viewDidLoad()
+    self.vm.inputs.onePassword(isAvailable: true)
+
+    self.onePasswordButtonHidden.assertValues([false])
+  }
+
+  func testTrackingEventsIfOnePassword_IsAvailable() {
+
+    self.vm.inputs.viewDidLoad()
+    self.vm.inputs.onePassword(isAvailable: true)
+
+    XCTAssertEqual([true, nil],
+                   self.trackingClient.properties(forKey: "1password_extension_available", as: Bool.self))
+
+    XCTAssertEqual([nil, true],
+                   self.trackingClient.properties(forKey: "one_password_extension_available", as: Bool.self))
+  }
+
+  func testPasswordText_OnePassword() {
+
+    self.vm.inputs.onePasswordFound(password: "123456")
+    self.passwordText.assertValues(["123456"])
+  }
+
+  func testEmailText_AfterFetchingUsersEmail() {
```

Comment:
The name of this test seems to imply that its just testing that the current email is fetched and correctly triggers a signal, but the implementation of the test seems to test the `passwordFieldDidTapGo` input, which I think is associated with submitting a new email. Does this test need to be renamed?

Reply:
The name of the test is correct. I had used the `passwordFieldDidGo` input because it's one of the inputs that triggers the signal. But it's a good idea to replace it with `self.vm.inputs.viewDidLoad()` (since I'm merging both), so it will avoid any confusion.